### PR TITLE
py / java sol for Firehose

### PR DESCRIPTION
**Pull request by freakin23**
_Place an "x" in the corresponding checkbox if it is done or does not apply to this pull request._

- [x] I have tested my code.
- [x] I have added my solution according to the steps [here](https://usaco.guide/general/adding-solution#steps).
- [x] I have followed the code conventions mentioned [here](https://usaco.guide/general/adding-solution/#code-conventions).
  - I understand that if it is clear that I have not attempted to follow these conventions, my PR will be closed.
  - If changes are requested, I will re-request a review after addressing them.
- [ ] I have linked this PR to any issues that it closes.

### CODE REVIEW

**Comment by freakin23** [comment]
AC (Python): https://dmoj.ca/submission/6615708
AC (Java): https://dmoj.ca/submission/6615710

File: solutions/silver/ccc-Firehose.mdx [hunk 1]
```diff
@@ -87,4 +87,148 @@ int main() {
...
+        for (int i = 0; i < numHouses; i++) {
+            houses[i] = io.nextInt();
+        }
+        
+        int numHydrants = io.nextInt();
+        
+        Arrays.sort(houses);
+
+        // Binary search for the lowest hose length
+        int left = 0, right = STREET_SIZE, ans = 0;
```
Comment (SansPapyrus683) [inline solutions/silver/ccc-Firehose.mdx:150]: declarations on separate lines

Comment (freakin23) [reply solutions/silver/ccc-Firehose.mdx]: for all three

Comment (SansPapyrus683) [reply solutions/silver/ccc-Firehose.mdx]: ye

Comment (freakin23) [reply solutions/silver/ccc-Firehose.mdx]: done

Comment (freakin23) [reply solutions/silver/ccc-Firehose.mdx]: How foolish of me! I thought I was making changes in the Java section but didn't realise I was keeping making changes in the C++ section.😂

File: solutions/silver/ccc-Firehose.mdx [hunk 2]
```diff
@@ -87,4 +88,147 @@ int main() {
...
+
+```java
+import java.io.*;
+import java.util.*;
+
+public class Firehose {
+	static final int MAX_H = 1000;
+	static final int STREET_SIZE = 1000000;
+
+	static boolean possible(int length, int numHouses, int numHydrants, int[] houses) {
```
Comment (TheGamingMousse) [inline solutions/silver/ccc-Firehose.mdx:101]: use `snake_case` for variable names

File: solutions/silver/ccc-Firehose.mdx [hunk 3]
```diff
@@ -87,4 +88,147 @@ int main() {
...
+
+		# The house that we need to connect a fire hydrant to.
+		start = houses[i]
+
+		for j in range(1, num_houses):
+			# Address of house at index j.
+			end = houses[(i + j) % num_houses]
+
+			"""
+            If the distance between the start and end houses is greater than
```
Comment (TheGamingMousse) [inline solutions/silver/ccc-Firehose.mdx:193]: are python block comments j weird like this or is it possible to make the indentation nicer

File: solutions/silver/ccc-Firehose.mdx [hunk 4]
```diff
@@ -87,4 +88,147 @@ int main() {
...
+houses = []
+for _ in range(num_houses):
+	houses.append(int(input()))
+
+num_hydrants = int(input())
+
+houses.sort()
+
+# Binary search for the lowest hose length
+l, r, ans = 0, STREET_SIZE, 0
```
Comment (TheGamingMousse) [inline solutions/silver/ccc-Firehose.mdx:220]: maybe im not familiar enough with python but this initialization was initially a bit confusing

consider breaking it up into multiple lines

File: solutions/silver/ccc-Firehose.mdx (hunk 2 above)
Comment (SansPapyrus683) [reply solutions/silver/ccc-Firehose.mdx:100]: this is java actually so

Comment (SansPapyrus683) [inline solutions/silver/ccc-Firehose.mdx:100]: why have a `numHouses` argument if you can just have `houses.length`?

File: solutions/silver/ccc-Firehose.mdx [hunk 5]
```diff
@@ -87,4 +88,147 @@ int main() {
...
+</JavaSection>
+
+<PySection>
+
+```py
+MAX_H = 1000
+STREET_SIZE = 1000000
+
+
+def possible(length: int, num_houses: int, num_hydrants: int, houses: list) -> bool:
```
Comment (SansPapyrus683) [inline solutions/silver/ccc-Firehose.mdx:175]: same here

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]

**Review by ryanchou-dev — APPROVED** [review]
looks fine after you address kevin's and justin's changes

File: solutions/silver/ccc-Firehose.mdx (hunk 3 above)
Comment (freakin23) [reply solutions/silver/ccc-Firehose.mdx]: sometimes it happens bcz of the bot.

File: solutions/silver/ccc-Firehose.mdx (hunk 5 above)
Comment (freakin23) [reply solutions/silver/ccc-Firehose.mdx:175]: i don't think it is a problem bcz either you pass it everytime or calculate it, there is not much difference. unless you make it a global

Comment (TheGamingMousse) [reply solutions/silver/ccc-Firehose.mdx:175]: it obviously isn’t a performance issue; his comment is that it is unnecessary, so there isn’t a point in having it

Comment (freakin23) [reply solutions/silver/ccc-Firehose.mdx:175]: okay i will change it

**Comment by freakin23** [comment]
AC (Java): https://dmoj.ca/submission/6616956
AC (Python): https://dmoj.ca/submission/6616958

File: solutions/silver/ccc-Firehose.mdx (hunk 2 above)
Comment (freakin23) [reply solutions/silver/ccc-Firehose.mdx]: done

**Review by SansPapyrus683 — APPROVED** [review]

**Pull request merged**